oozie-to-airflow/mappers/mapreduce_mapper.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Maps Oozie pig node to Airflow's DAG"""
+import os
+from typing import Set, Dict
+from xml.etree.ElementTree import Element
+
+import jinja2
+from airflow.utils.trigger_rule import TriggerRule
+
+from definitions import ROOT_DIR
+from mappers.action_mapper import ActionMapper
+from mappers.prepare_mixin import PrepareMixin
+from utils import el_utils, xml_utils
+from utils.file_archive_extractors import FileExtractor, ArchiveExtractor
+
+
+# pylint: disable=too-many-instance-attributes
+class MapReduceMapper(ActionMapper, PrepareMixin):
+    """
+    Converts a MapReduce Oozie node to an Airflow task.
+    """
+
+    properties: Dict[str, str]
+    params_dict: Dict[str, str]
+
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        params=None,
+        template_file_name: str = "mapreduce.tpl",
+        **kwargs,
+    ):
+        ActionMapper.__init__(self, oozie_node=oozie_node, name=name, trigger_rule=trigger_rule, **kwargs)
+        if params is None:
+            params = dict()
+        self.template = template_file_name
+        self.params = params
+        self.task_id = name
+        self.trigger_rule = trigger_rule
+        self.properties = {}
+        self.params_dict = {}
+        self.file_extractor = FileExtractor(oozie_node=oozie_node, params=params)
+        self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, params=params)
+        self._parse_oozie_node()
+
+    def _parse_oozie_node(self):
+        name_node_text = self.oozie_node.find("name-node").text
+        self.name_node = el_utils.replace_el_with_var(name_node_text, params=self.params, quote=False)
+        self._parse_config()
+        self._parse_params()
+        self.files, self.hdfs_files = self.file_extractor.parse_node()
+        self.archives, self.hdfs_archives = self.archive_extractor.parse_node()
+
+    def _parse_params(self):
+        param_nodes = xml_utils.find_nodes_by_tag(self.oozie_node, "param")
+        if param_nodes:
+            self.params_dict = {}
+            for node in param_nodes:
+                param = el_utils.replace_el_with_var(node.text, params=self.params, quote=False)
+                key, value = param.split("=")
```

Comment:
```suggestion
                key, value = param.split("=", 1)
```
In some cases, this can cause problems.

Reply:
Done